lib/ansible/modules/packaging/os/yum.py

```diff
@@ -1461,6 +1461,15 @@ def run(self):
         if error_msgs:
             self.module.fail_json(msg='. '.join(error_msgs))
 
+        if self.update_cache and not self.names and not self.list:
+            rc, stdout, stderr = self.module.run_command(self.yum_basecmd + ['clean', 'expire-cache'])
+            self.module.exit_json(
```

Comment:
Is there a way to tell if cache update was not successful and let the user know?

Reply:
@mkrizek I think so, I probably should have done that to begin with. I'll come up with a failure scenario to test and will update the patch accordingly. Thanks